src/pages/patientView/trialMatch/TrialMatchTable.tsx

```diff
@@ -0,0 +1,321 @@
+                                    ))}
+                                </ul>
+                            ) : (
+                                <span>{`${genomicGroupMatch.matches[0].trueHugoSymbol} ${genomicGroupMatch.matches[0].trueProteinChange} `}
+                                    {genomicGroupMatch.matches[0].sampleIds.map((sampleId: string) => (
+                                        <span className={styles.genomicSpan}>
+                                                {this.props.sampleManager!.getComponentForSample(sampleId, 1, '')}
+                                            </span>
+                                    ))}
+                                </span>
+                            )}
+                        </span>
+                    )}
+                    </div>
+                ))}
+                { clinicalGroupMatch.notMatches.length > 0 &&
+                    <DefaultTooltip
+                        placement='bottomLeft'
+                        trigger={['hover', 'focus']}
+                        overlay={this.tooltipGenomicContent(clinicalGroupMatch.notMatches)}
+                        destroyTooltipOnHide={false}
+                        onPopupAlign={placeArrowBottomLeft}
+                    >
+                        {this.mainContent(`No alterations in ${this.getHugoSymbolName(clinicalGroupMatch.notMatches, 3)} defined by the trial`)}
+                    </DefaultTooltip>
+                }
+            </span>
+        );
+    }
+
+    public tooltipGenomicContent(data: Array<IGenomicGroupMatch>) {
+        return (
+            <div className={styles.tooltip}>
+                {data.map((genomicGroupMatch: IGenomicGroupMatch) => (
+                    <div className={styles.genomicSpan}><b>Not </b>{`${genomicGroupMatch.genomicAlteration.replace(/!/g, '')} `}
+                        {genomicGroupMatch.matches[0].sampleIds.map((sampleId: string) => (
+                            <span className={styles.genomicSpan}>
+                                {this.props.sampleManager!.getComponentForSample(sampleId, 1, '')}
+                            </span>
+                        ))}
+                    </div>
+                ))}
+            </div>
+        );
+    }
+    public tooltipClinicalContent(data: Array<string>) {
+        return (
+            <div className={styles.tooltip}>
+                <ul className={styles.alterationUl}>
+                    {data.map((cancerType: string) => (
+                        <li>{cancerType}</li>
+                    ))}
+                </ul>
+            </div>
+        );
+    }
+
+    public mainContent(content: string) {
+        return (
+            <span>
+                <a>{content}</a>
+            </span>
+        );
+    }
+
+    public getHugoSymbolName(matches: Array<IGenomicGroupMatch>, threshold: number) {
+        const hugoSymbolSet = new Set([...matches].map(x => x.genomicAlteration.split(' ')[0].slice(1)));
```

Comment:
i think the only reason to use Set here is if this is meant to be a unique collection.  Set will automatically enforce uniqueness.  Is that required, if so, we should use Set.  If not, use Array.

Reply:
Yes, I want to have an unique collection of hugo_symbols so I use Set at here.